presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -229,6 +229,15 @@ default int compareTo(int leftPosition, int leftOffset, int leftLength, Block ri
      */
     boolean isNull(int position);
 
+    /**
+     * Create a new block from the current block by keeping the same elements
+     * only with respect to {@code visiblePositions}.
+     */
+    default Block mask(int[] visiblePositions)
```

Comment:
lets name this `getPositions` and update the documentation to say "may return a view over the data in this block or may return a copy".  This keeps the naming consistent with `getRegion` and `copyRegion`.

Reply:
BTW the `int[]` argument type is fine, we are looking at making a similar change to `copyPositions`